actors/builtin/miner/miner_actor.go

```diff
@@ -3108,3 +3125,114 @@ func checkPeerInfo(rt Runtime, peerID abi.PeerID, multiaddrs []abi.Multiaddrs) {
 		rt.Abortf(exitcode.ErrIllegalArgument, "multiaddr size of %d exceeds maximum of %d", totalSize, MaxMultiaddrData)
 	}
 }
+
+type ChangeBeneficiaryParams struct {
+	NewBeneficiary addr.Address
+	NewQuota       abi.TokenAmount
+	NewExpiration  abi.ChainEpoch
+}
+
+// Do necessary parameter check for ChangeBeneficiary if NewBeneficiary is not owner
+//		NewQuota has to be a positive number
+//		newExpiration has to be greater than currEpoch height
+func checkNewBeneficialParam(rt Runtime, params *ChangeBeneficiaryParams) {
+	if params.NewExpiration <= rt.CurrEpoch() {
+		rt.Abortf(exitcode.ErrIllegalArgument, "new beneficial expire (%d) date must bigger than current epoch (%d)", params.NewExpiration, rt.CurrEpoch())
+	}
+
+	if params.NewQuota.LessThanEqual(big.NewInt(0)) {
+		rt.Abortf(exitcode.ErrIllegalArgument, "beneficial quota (%s) should bigger than zero", params.NewQuota)
+	}
+}
+
+// ChangeBeneficiary proposal/approve beneficiary change
+func (a Actor) ChangeBeneficiary(rt Runtime, params *ChangeBeneficiaryParams) *abi.EmptyValue {
+	if params == nil {
+		rt.Abortf(exitcode.ErrIllegalArgument, "empty ChangeBeneficiary Paramenters")
+	}
+	if params.NewBeneficiary.Protocol() != addr.ID {
+		rt.Abortf(exitcode.ErrIllegalArgument, "NewBeneficiary address must be an ID address")
+	}
+
+	var st State
+	rt.StateTransaction(&st, func() {
+		info := getMinerInfo(rt, &st)
+
+		// This is a ChangeBeneficiary proposal when the caller is Owner
+		if rt.Caller() == info.Owner {
+			rt.ValidateImmediateCallerIs(info.Owner)
+			if params.NewBeneficiary != info.Owner {
+				// No need to check others when the newBeneficiary is set back to owner
+				checkNewBeneficialParam(rt, params)
+			}
+			if info.Beneficiary != info.Owner && info.BeneficiaryInfo.Effective(rt.CurrEpoch()) {
```

Comment:
Comment: "If there is a current beneficiary, store the new proposal and set next approver to that current beneficiary".

If you take my suggestion to add approval flags rather than a strict sequence, this case will merge with the one below.